assets/watch-lists/components/WatchListSchedule.jsx

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {get} from 'lodash';
+import moment from 'moment';
+
+import { gettext } from 'utils';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+
+import SelectInput from 'components/SelectInput';
+
+class WatchListSchedule extends React.Component {
+    constructor(props) {
+        super(props);
+        
+        this.options = [
+            {
+                value: 'immediate',
+                'text': 'Immediate'
+            },
+            {
+                value: 'two_hour',
+                'text': 'Every Two Hours'
+            },
+            {
+                value: 'four_hour',
+                'text': 'Every Four Hours'
+            },
+            {
+                value: 'daily',
+                'text': 'Every day at'
+            },
+            {
+                value: 'weekly',
+                'text': 'Every week at'
+            },
+        ];
+
+        this.onChangeSchedule = this.onChangeSchedule.bind(this);
+        this.onTimeChange = this.onTimeChange.bind(this);
+
+        this.state = {needTime: this.getNeedTime()};
+    }
+
+    componentDidUpdate(prevPops) {
+        if (get(this.props, 'watchList._id') !== get(prevPops, 'watchList._id')) {
+            this.setState({ needTime: this.getNeedTime() });
+        }
+    }
+
+    getNeedTime(option=get(this.props, 'watchList.schedule.interval')) {
```

Comment:
It's better to pass this value when calling this function instead of using `this.props` in the function signature as the default value